src/hooks/useGallery/useGallery.tsx

```diff
@@ -2,6 +2,13 @@ import * as React from 'react';
 
 import {GalleryContext} from './GalleryContext';
 
-export function useGallery() {
-    return React.useContext(GalleryContext);
+export function useGallery(onCloseCallback?: () => void) {
+    const {openGallery, closeGallery: contextClose} = React.useContext(GalleryContext);
```

Comment:
Maybe returning `isOpen` flag from the GaleryContext is suitable, instead of transmitting a callback?